### PR TITLE
Revived submitBatch

### LINKED ISSUE #34: Bring back submit batch contract interactor
https://github.com/thehubbleproject/hubble-commander/blob/master/core/bazooka.go#L419

### PR DESCRIPTION
**Pull request by vaibhavchellani**
Fixes #34

### CODE REVIEW

File: core/bazooka.go [hunk 1]
```diff
@@ -337,6 +337,90 @@ func (b *Bazooka) DecodeState(stateBytes []byte) (ID, balance, nonce, token *big
...
+		compressedTxs, err := b.CompressTxs(commitment.Txs)
+		if err != nil {
+			b.log.Error("Unable to compress txs", "error", err)
+			return err
+		}
+		txs = append(txs, compressedTxs)
+		updatedRoots = append(updatedRoots, commitment.UpdatedRoot)
+		totalTxs += len(commitment.Txs)
+
+		// TODO cleanup
```
Comment (ChihChengLiang) [inline core/bazooka.go:368]: Do we want to create an issue that adds an AggregatedSignature wrapper with a method like `ToSol()`?

File: core/bazooka.go [hunk 2]
```diff
@@ -337,6 +337,90 @@ func (b *Bazooka) DecodeState(stateBytes []byte) (ID, balance, nonce, token *big
...
+		aggregatedSigBigInt := [2]*big.Int{sig1bigInt, sig2bigInt}
+		aggregatedSig = append(aggregatedSig, aggregatedSigBigInt)
+	}
+
+	b.log.Info("Batch prepared", "totalTransactions", totalTxs)
+
+	rollupAddress := ethCmn.HexToAddress(config.GlobalCfg.RollupAddress)
+	stakeAmount := big.NewInt(0)
+
+	// TODO fix
```
Comment (ChihChengLiang) [inline core/bazooka.go:384]: What's the thing we want to fix here? Is it adding a feeReceiver to the config?

File: core/bazooka.go [hunk 3]
```diff
@@ -412,115 +496,10 @@ func (b *Bazooka) FireDepositFinalisation(TBreplaced UserState, siblings []UserS
...
-	// // err = DBInstance.LogBatch(0, txs[0].Type, updatedRoot.String(), compressedTxs)
-	// // if err != nil {
-	// // 	return err
-	// // }
-
 	return nil
 }
 
-func (b *Bazooka) GenerateAuthObj(client *ethclient.Client, callMsg ethereum.CallMsg) (auth *bind.TransactOpts, err error) {
+func (b *Bazooka) generateAuthObj(client *ethclient.Client, callMsg ethereum.CallMsg) (auth *bind.TransactOpts, err error) {
```
Comment (ChihChengLiang) [inline core/bazooka.go:502]: We change the MixCaps to camelCase here, is the intention that we don't want to export generateAuthObj in the package?

File: core/bazooka.go [hunk 4]
```diff
@@ -337,6 +337,90 @@ func (b *Bazooka) DecodeState(stateBytes []byte) (ID, balance, nonce, token *big
...
+		sig2bigInt := big.NewInt(0)
+		sig2bigInt.SetBytes(sig2)
+		aggregatedSigBigInt := [2]*big.Int{sig1bigInt, sig2bigInt}
+		aggregatedSig = append(aggregatedSig, aggregatedSigBigInt)
+	}
+
+	b.log.Info("Batch prepared", "totalTransactions", totalTxs)
+
+	rollupAddress := ethCmn.HexToAddress(config.GlobalCfg.RollupAddress)
+	stakeAmount := big.NewInt(0)
```
Comment (ChihChengLiang) [inline core/bazooka.go:382]: We should create an issue to make stakeAmount configurable too.

**Review by ChihChengLiang — APPROVED** [review]
Looks good to me. Ask some questions and suggest issues to create.

File: core/bazooka.go (hunk 1 above)
Comment (vaibhavchellani) [reply core/bazooka.go:368]: https://github.com/thehubbleproject/hubble-commander/issues/56

File: core/bazooka.go (hunk 2 above)
Comment (vaibhavchellani) [reply core/bazooka.go:384]: Right now I have added 0 as fee receiver to fix build but we need to replace that with the correct one.

Lets talk here: https://github.com/thehubbleproject/hubble-commander/issues/47

File: core/bazooka.go (hunk 3 above)
Comment (vaibhavchellani) [reply core/bazooka.go:502]: Bingo!

**Pull request merged**